rbtcwrapperproxy/RBTCWrapperProxy.sol

```diff
@@ -183,36 +186,49 @@ contract RBTCWrapperProxy {
-        require(successOfApprove, "Failed to approve converter to transfer WRBTC");
+        IWrbtcERC20(wrbtcTokenAddress).deposit.value(_reserveAmounts[0])();
+        success = IWrbtcERC20(wrbtcTokenAddress).approve(_liquidityPoolConverterAddress, _reserveAmounts[0]);
+        require(success, "Failed to approve converter to transfer WRBTC");
         
         for (uint256 i = 1; i < _reserveTokens.length; i++) {
             IERC20Token reserveToken = _reserveTokens[i];
             uint256 reserveAmount = _reserveAmounts[i];
-            successOfTransfer = IERC20Token(reserveToken).transferFrom(msg.sender, address(this), reserveAmount);
-            require(successOfTransfer, "Failed to transfer reserve token from user");
-            successOfApprove = IERC20Token(reserveToken).approve(_liquidityPoolConverterAddress, reserveAmount);
-            require(successOfApprove, "Failed to approve converter to transfer reserve token");
+            success = IERC20Token(reserveToken).transferFrom(msg.sender, address(this), reserveAmount);
+            require(success, "Failed to transfer reserve token from user");
+            success = IERC20Token(reserveToken).approve(_liquidityPoolConverterAddress, reserveAmount);
+            require(success, "Failed to approve converter to transfer reserve token");
         }
 
         ILiquidityPoolV1Converter _liquidityPoolConverter = ILiquidityPoolV1Converter(_liquidityPoolConverterAddress);
         ISmartToken _poolToken = ISmartToken(address(_liquidityPoolConverter.token()));
         uint256 poolTokenAmountBefore = _poolToken.balanceOf(address(this));
         _liquidityPoolConverter.addLiquidity(_reserveTokens, _reserveAmounts, _minReturn);
-        uint256 poolTokenAmountAfter = _poolToken.balanceOf(address(this));
-        uint256 poolTokenAmount = poolTokenAmountAfter.sub(poolTokenAmountBefore);
+        uint256 poolTokenAmount = _poolToken.balanceOf(address(this)).sub(poolTokenAmountBefore);
         
-        successOfTransfer = _poolToken.transfer(msg.sender, poolTokenAmount);
-        require(successOfTransfer, "Failed to transfer pool token to user");
+        success = _poolToken.transfer(msg.sender, poolTokenAmount);
+        require(success, "Failed to transfer pool token to user");
+
+        (uint256 rsvBalance, , , , ) = _liquidityPoolConverter.reserves(wrbtcTokenAddress);
+        uint256 costReserveAmount = ISovrynSwapFormula(addressOf(SOVRYNSWAP_FORMULA)).fundCost(_poolToken.totalSupply(), rsvBalance, _liquidityPoolConverter.reserveRatio(), poolTokenAmount);
+        if (_reserveAmounts[0] > costReserveAmount) {
+            IWrbtcERC20(wrbtcTokenAddress).withdraw(_reserveAmounts[0].sub(costReserveAmount));
+            (success,) = msg.sender.call.value(_reserveAmounts[0].sub(costReserveAmount))("");
+            require(success, "Failed to send extra RBTC back to user");
+        }
```

Comment:
consider moving RBTC block to the end of the function to prevent reentrancy

Reply:
@wukong-xxs please address this before deployment